### PR TITLE
Prevent false positives for pages whose CSP cannot be checked

### PR DESCRIPTION
**Pull request by juliettepretot**
# Checklist

- [ ] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by
        [Cloudbuild](/cloudbuild.yaml)
  - [ ] I have updated [documentation](/docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

File: chrome_extension/background.js [hunk 1]
```diff
@@ -19,21 +19,31 @@
 const GREEN_ICON_PATH = './icon-green.png';
 const RED_ICON_PATH = './icon-red.png';
 
-const isUsingDesiredCspPerTab = new Map();
+// Map of tabs that are applying the desired CSP. The key is the `tabId`, the
+// value is the specific `url` that applies them. Keeping track of the URL is
+// necessary since administrators can block access to the `chrome.webRequest`
+// API used to check CSP on a per page basis. Without checking that the URL
+// matches, a user could navigate from a secure page to a potentially
```
Comment (conradgrobler) [inline chrome_extension/background.js:26]: How does the user navigate to a different page?

Is it only if the user types a URL in the address bar, or can a user click on a link to also navigate or is this blocked by the CSP?

**Review by conradgrobler — APPROVED** [review]
Interesting case.

File: chrome_extension/background.js (hunk 1 above)
Comment (juliettepretot) [reply chrome_extension/background.js:26]: For this case form of navigation works, clicking a bookmark, typing a URL etc. As long as it is in the same tab.

**Review by tiziano88 — APPROVED** [review]

File: chrome_extension/background.js (hunk 1 above)
Comment (juliettepretot) [reply chrome_extension/background.js:26]: RE disabling in page navigation, it prevents some, unfortunately not all. I created an issue to tackle this: https://github.com/project-oak/oak/issues/1792

**Pull request merged**